app/Platform/Controllers/GameController.php

```diff
@@ -86,76 +82,23 @@ public function index(GameListRequest $request): InertiaResponse
         return Inertia::render('games', $props);
     }
 
-    public function popular(): void
-    {
-        $this->authorize('viewAny', $this->resourceClass());
-
-        // return view('game.popular');
-    }
-
-    public function create(): void
-    {
-        $this->authorize('store', $this->resourceClass());
-    }
-
-    public function store(Request $request): void
-    {
-        $this->authorize('store', $this->resourceClass());
-    }
-
-    public function show(Request $request, Game $game, ?string $slug = null): View|RedirectResponse
-    {
+    public function show(
+        Request $request,
+        Game $game,
+        LoadGameWithRelationsAction $loadGameWithRelationsAction,
+        BuildGameShowPagePropsAction $buildGameShowPagePropsAction
+    ): InertiaResponse {
         $this->authorize('view', $game);
 
-        if (!$this->resolvesToSlug($game->slug, $slug)) {
-            return redirect($game->canonicalUrl);
-        }
-
-        /**
-         * add user context
-         */
-        $playerGame = $request->user() ? $request->user()->games()->where('games.id', $game->id)->first() : null;
-
-        $game = $playerGame ?? $game;
-
-        $game->load([
-            'achievements' => function ($query) use ($playerGame) {
-                // $query->with('game');
-
-                // $query->published();
-
-                /*
-                 * add user context
-                 */
-                if ($playerGame) {
-                    $query->withUnlocksByUser(request()->user());
-                    $query->orderByDesc('unlocked_hardcore_at');
-                    $query->orderByDesc('unlocked_at');
-                }
-            },
-            'leaderboards',
-            'forumTopic',
-        ]);
-
-        // $game->achievements->each->setRelation('game', $game);
-
-        return view($this->resourceName() . '.show')->with('game', $game);
-    }
-
-    public function edit(Game $game): View
-    {
-        $this->authorize('update', $game);
-
-        return view($this->resourceName() . '.edit')->with('game', $game);
-    }
+        /** @var ?User $user */
+        $user = $request->user();
 
-    public function update(GameRequest $request, Game $game): RedirectResponse
-    {
-        $this->authorize('update', $game);
+        $game = $loadGameWithRelationsAction->execute($game, AchievementFlag::OfficialCore);
+        $props = $buildGameShowPagePropsAction->execute($game, $user);
```

Comment:
Cannot view `/game2/` for a game without achievements.
http://localhost:64000/game2/26960
http://localhost:64000/game2/32412

Or maybe it's something else? I get the same error trying to view the `/game2/` for any of the "New Releases" games on the main page.

Reply:
It seems `created_at` and `updated_at` for sets is nullable. This may be worth investigating in a future PR, for now the pages that are impacted won't crash anymore.